hw/ip/otbn/dv/rig/rig/gens/known_wdr.py

```diff
@@ -0,0 +1,109 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+from typing import Optional
+
+from shared.operand import ImmOperandType, RegOperandType
+from shared.insn_yaml import InsnsFile
+
+from ..config import Config
+from ..model import Model
+from ..program import ProgInsn, Program
+
+from ..snippet import ProgSnippet
+from ..snippet_gen import GenCont, GenRet, SnippetGen
+
+
+class KnownWDR(SnippetGen):
+    '''A snippet generator that generates known values (all zeros or all ones
+    for now) for WDRs.
+
+    '''
+
+    def __init__(self, cfg: Config, insns_file: InsnsFile) -> None:
+        super().__init__()
+
+        if 'bn.xor' not in insns_file.mnemonic_to_insn:
+            raise RuntimeError('BN.XOR instruction not in instructions file')
+
+        if 'bn.not' not in insns_file.mnemonic_to_insn:
+            raise RuntimeError('BN.NOT instruction not in instructions file')
+
+        self.insn_xor = insns_file.mnemonic_to_insn['bn.xor']
+        self.insn_not = insns_file.mnemonic_to_insn['bn.not']
```

Comment:
There's a handy wrapper for this in the base class called `_get_named_insn` (see e.g. the `Loop` generator for how to use it).